src/hooks/coreSDK/safeCoreSDK.ts

```diff
@@ -95,15 +73,15 @@ export const initSafeSDK = async ({
   }
 
   if (undeployedSafe) {
-    return Safe.create({
-      ethAdapter: createReadOnlyEthersAdapter(provider),
+    return Safe.init({
+      provider: provider._getConnection().url,
```

Comment:
Looking into this further I think whenever we call `_getConnection()` it actually does a fetch request. I will try to find another way to get the URL.

Reply:
Can confirm that providing a real RPC URL for the tests does indeed make requests to that RPC so the fetches are not mocked. Need to figure this out before we can merge this.